homeassistant/auth/providers/trusted_networks.py

```diff
@@ -86,40 +89,60 @@ def support_mfa(self) -> bool:
         """Trusted Networks auth provider does not support MFA."""
         return False
 
+    @callback
+    def is_allowed_request(self) -> bool:
+        """Return if it is an allowed request."""
+        request = http.current_request.get()
+        if request is not None and (
+            self.hass.http.use_x_forwarded_for
+            or hdrs.X_FORWARDED_FOR not in request.headers
+        ):
```

Comment:
Can this be reformulated/split a bit. My head has a heard time with all the nots,ands and ors.

Reply:
Perhaps even Invert the function to is_blocked_request since that is how it's used.

Reply:
I think in the end, the question is; should we even ever process forwarded-for requests, without a reverse proxy being configured at all...

I think a warning and getting this into the beta is fine for now, lets re-evaluate the handling for the next release.

Reply:
I wasn't against the warning. I was against the hard to read logic with many negations.

Reply:
Yes, I agree, but I think it is good to have in the beta. Given the time is close, we should improve later.

I'm opening a PR as a follow-up, that moves it out of here, into the HTTP server directly.
If that is an acceptable solution, this can be revered.

Reply:
Alternative approach in #51332